### PR TITLE
feat(`anvil`): `wallet_` namespace + inject P256BatchDelegation + executor

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Introduce `odyssey` features to anvil, activated using the `--odyssey`.

## Solution

- [x] `wallet_getCapabilities`
- [x] `wallet_sendTransaction`
- [x]  Inject P256BatchDelegation in backend
- [x]  Setup funded executor that sponsors the transactions
- [x] Tests

### CODE REVIEW

File: crates/anvil/core/src/eth/wallet.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+///
+/// The sequencer will only perform delegations, and act on behalf of delegated accounts, if the
+/// account delegates to one of the addresses specified within this capability.
+///
+/// [eip-7702]: https://eips.ethereum.org/EIPS/eip-7702
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub struct DelegationCapability {
+    /// A list of valid delegation contracts.
+    pub addresses: Vec<Address>,
+}
```
Comment (mattsse) [inline crates/anvil/core/src/eth/wallet.rs:14]: all of these should be moved to alloy, but can do that once that pr is merged and published

File: crates/anvil/src/eth/backend/mem/mod.rs [hunk 2]
```diff
@@ -111,6 +116,17 @@ pub mod storage;
...
...10df9565b039060025afa156104f9576101dd915f51610ce3565b505050505f90565b50505050505f90565b610f2b9150610f1e610d616106e492610cc2565b6080850151151590610f31565b5f610e1f565b906001600160f81b0319600160f81b831601610f955780610f85575b610f8057601f60fb1b600160fb1b821601610f69575b50600190565b600160fc1b90811614610f7c575f610f63565b5f90565b505f90565b50600160fa1b8181161415610f4d565b50505f90565b80519282515f5b858110610fb457505050505050600190565b8083018084116105b5578281101561100757610fe56001600160f81b0319610fdc8488610cd2565b51169187610cd2565b516001600160f81b03191603610ffd57600101610fa2565b5050505050505f90565b505050505050505f90565b80516060929181611021575050565b9092506003600284010460021b604051937f4142434445464748494a4b4c4d4e4f505152535455565758595a616263646566601f527f6768696a6b6c6d6e6f707172737475767778797a303132333435363738392d5f603f52602085019282860191602083019460208284010190600460038351955f85525b0191603f8351818160121c16515f538181600c1c1651600153818160061c165160025316516003535f5181520190878210156110db5760049060039061109a565b5095505f93600393604092520160405206600204809303613d3d60f01b81525203825256fea26469706673582212200ba93b78f286a25ece47e9403c47be9862f9b8b70ba1a95098667b90c47308b064736f6c634300081a0033");
```
Comment (mattsse) [inline crates/anvil/src/eth/backend/mem/mod.rs:125]: I think we should make this configurable or move to a separate file

**Review by mattsse — COMMENTED** [review]
ptal @klkvr

File: crates/anvil/src/eth/backend/mem/mod.rs (hunk 2 above)
Comment (klkvr) [reply crates/anvil/src/eth/backend/mem/mod.rs:125]: yeah would be nice to have an RPC method to set those parameters I think

Comment (yash-atreya) [reply crates/anvil/src/eth/backend/mem/mod.rs:125]: Added `anvil_addCapability`, which adds an address to the list of valid delegations. 

Users can deploy their delegation contracts to anvil and call `anvil_addCapability(deployed_address)`.

**Comment by zerosnacks** [comment]
General question: I can't find `wallet_sendTransaction` in the [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792) spec. Is this a custom API of Odyssey?

Possibly related to this restriction: https://github.com/ithacaxyz/odyssey/blob/bc8c87e08ca07dfec3a6cc00bb0488c1e24248c4/crates/wallet/src/lib.rs#L12-L13

It appears we are missing `wallet_sendCalls`, `wallet_getCallsStatus`, `wallet_showCallsStatus` that is supported by Wagmi here: https://wagmi.sh/react/api/hooks/useCallsStatus

**Comment by yash-atreya** [comment]
> General question: I can't find `wallet_sendTransaction` in the [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792) spec. Is this a custom API of Odyssey?
> 
> Possibly related to this restriction: https://github.com/ithacaxyz/odyssey/blob/bc8c87e08ca07dfec3a6cc00bb0488c1e24248c4/crates/wallet/src/lib.rs#L12-L13
> 
> It appears we are missing `wallet_sendCalls`, `wallet_getCallsStatus`, `wallet_showCallsStatus` that is supported by Wagmi here: https://wagmi.sh/react/api/hooks/useCallsStatus

Yes `wallet_sendTransaction` is specific to odyssey and not in the EIP

**Comment by zerosnacks** [comment]
> Yes `wallet_sendTransaction` is specific to odyssey and not in the EIP

Opened a related ticket for it here to discuss an `odyssey_` namespace: https://github.com/ithacaxyz/odyssey/issues/50

Not a blocker for this PR

**Review by klkvr — APPROVED** [review]
lgtm! this duplicates a bunch of odyssey logic but should be fine for now

hopefully we can clean this up with reth-anvil soon :)

**Pull request merged**